apps/docs/content/guides/auth/social-login/auth-google.mdx

```diff
@@ -516,66 +171,473 @@ Before you can use Sign in with Google, you need to obtain a [Google Cloud Platf
+
+1. Load the Google client library in your app by including the third-party script:
+
+   ```html
+   <script src="https://accounts.google.com/gsi/client" async></script>
+   ```
+
+1. Use the [HTML Code Generator](https://developers.google.com/identity/gsi/web/tools/configurator) to customize the look, feel, features and behavior of the Sign in with Google button.
+1. Pick the _Swap to JavaScript callback_ option, and input the name of your callback function. This function will receive a [`CredentialResponse`](https://developers.google.com/identity/gsi/web/reference/js-reference#CredentialResponse) when sign in completes.
+
+   To make your app compatible with Chome's third-party-cookie phase-out, make sure to set `data-use_fedcm_for_prompt` to `true`.
+
+   Your final HTML code might look something like this:
+
+   ```html
+   <div
+     id="g_id_onload"
+     data-client_id="<client ID>"
+     data-context="signin"
+     data-ux_mode="popup"
+     data-callback="handleSignInWithGoogle"
+     data-nonce=""
+     data-auto_select="true"
+     data-itp_support="true"
+     data-use_fedcm_for_prompt="true"
+   ></div>
+
+   <div
+     class="g_id_signin"
+     data-type="standard"
+     data-shape="pill"
+     data-theme="outline"
+     data-text="signin_with"
+     data-size="large"
+     data-logo_alignment="left"
+   ></div>
+   ```
+
+1. Create a `handleSignInWithGoogle` function that takes the `CredentialResponse` and passes the included token to Supabase. The function needs to be available in the global scope for Google's code to find it.
+
+   ```ts
+   async function handleSignInWithGoogle(response) {
+     const { data, error } = await supabase.auth.signInWithIdToken({
+       provider: 'google',
+       token: response.credential,
+     })
+   }
+   ```
+
+1. _(Optional)_ Configure a nonce. The use of a nonce is recommended for extra security, but optional. The nonce should be generated randomly each time, and it must be provided in both the `data-nonce` attribute of the HTML code and the options of the callback function.
+
+   ```ts
+   async function handleSignInWithGoogle(response) {
+     const { data, error } = await supabase.auth.signInWithIdToken({
+       provider: 'google',
+       token: response.credential,
+       nonce: '<NONCE>',
+     })
+   }
+   ```
+
+   Note that the nonce should be the same in both places, but because Supabase Auth expects the provider to hash it (SHA-256, hexadecimal representation), you need to provide a hashed version to Google and a non-hashed version to `signInWithIdToken`.
+
+   You can get both versions by using the in-built `crypto` library:
+
+   ```js
```

Comment:
Rewrote this to work browser-side since that seems to fit more with the example (script tags, functions in browser global scope). Would appreciate a second set of 👀 !

Reply:
I'm not sure I would class it as "browser side" since it's actually the olden days server side. But sure we can try it. I don't think I've seen anyone in Supabase use it.